extensions/amp-a4a/0.1/amp-ad-render.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {A4AVariableSource} from '../../amp-a4a/0.1/a4a-variable-source';
+import {Renderer} from './amp-ad-type-defs';
+import {createElementWithAttributes} from '../../../src/dom';
+import {dev} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {getDefaultBootstrapBaseUrl} from '../../../src/3p-frame';
+import {
+  installFriendlyIframeEmbed,
+  setFriendlyIframeEmbedVisible,
+} from '../../../src/friendly-iframe-embed';
+import {installUrlReplacementsForEmbed} from '../../../src/service/url-replacements-impl';
+import {setStyle} from '../../../src/style';
+
+/**
+ * Render a validated AMP creative directly in the parent page.
+ */
+export class FriendlyFrameRenderer extends Renderer {
```

Comment:
I am expecting at least three types of renderer. To avoid making the file too long. Can we separate each renderer implementation class to its own file? Thanks.

Reply:
Done.